libr/util/file.c

```diff
@@ -365,6 +365,7 @@ R_API char *r_file_slurp(const char *str, R_NULLABLE size_t *usz) {
 				}
 				size += r;
 			} while (!feof (fd));
+			buf[size] = '\0';
```

Comment:
change the code of the function to just do a single malloc(sz+1) if you want to null terminate the buffer. the current implementation do realloc for no reason.

Reply:
I believe it does `realloc` because the size is not known. If it was known, we would not enter this path.

Reply:
the size should be known if its a regular file, only edge case are the files in /proc i think. i didnt tried the code, but it was a blind guess and its unrelated to the PR

Reply:
yes indeed the general case is not this path. In general it just does `malloc(sz + 1)`, but this function tries to deal also with some files where it can't determine the size ahead of time, so continuous reallocation are needed.